app/Http/Controllers/Api/DeviceController.php

```diff
@@ -318,5 +322,69 @@ public function anyList(BcsRequest $request)
 
         return response()->json($ret_val);
     }
-
+    
+    /**
+     * @param \BComeSafe\Http\Requests\SaveDeviceRequest $request
+     *
+     * @return array
+     */
+    public function anyUpdateDevice(SaveDeviceRequest $request)
+    {
+        $update = array();
+        $params = $request->all();
+        //
+        if (isset($params['user_phone'])) {
```

Comment:
Could we use `if ($request->filled('user_phone'))` which checks for both `isset` and `!empty`? Would be shorter and that way we don't need to do `$params = $request->all();`.

Reply:
Worth to check, but risky to change now. Added "Todo".